src/praisonai/praisonai/cli/session/unified.py

```
@@ -140,6 +141,91 @@ def _get_session_path(self, session_id: str) -> Path:
     def _get_last_session_path(self) -> Path:
         """Get the path to the last session marker file."""
         return self.session_dir / ".last_session"
+
+    @staticmethod
+    def _messages_common_prefix(
+        left: List[Dict[str, str]], right: List[Dict[str, str]]
+    ) -> int:
+        """Return shared message prefix length for safe concurrent merge."""
+        prefix = 0
+        for left_msg, right_msg in zip(left, right):
```

Ruff (0.15.15)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)